pkg/utils/utils.go

```diff
@@ -61,3 +65,59 @@ func SendToSocket(socket string, command string) error {
 	}
 	return nil
 }
+
+// checkValidity runs a HAProxy configuration validity check on a file
+func checkValidity(configFile string) error {
+	out, err := exec.Command("haproxy", "-c", "-f", configFile).CombinedOutput()
+	if err != nil {
+		glog.Warningf("Error validating config file:\n%v", string(out))
+		return err
+	}
+	return nil
+}
+
+// multibinderERBOnly generates a config file from ERB template by invoking multibinder-haproxy-erb
+func multibinderERBOnly(configFile string) (string, error) {
+	_, err := exec.Command("multibinder-haproxy-erb", "/usr/local/sbin/haproxy", "-f", configFile, "--erb-write-only").CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	return configFile[:strings.LastIndex(configFile, ".erb")], nil
+}
+
+// RewriteConfigFiles safely replaces configuration files with new contents after validation
+func RewriteConfigFiles(data []byte, reloadStrategy, configFile string) error {
+	tmpf := "/etc/haproxy/new_cfg.erb"
```

Comment:
I was taking into account that, at this moment, `/` is the only filesystem. But we cannot say that this will stay the same about the tmp subdir. Since this is more a convention than an issue the actual implementation is fine.